apps/web/src/components/digest-demo-flow/NodeStep.tsx

```diff
@@ -8,6 +8,15 @@ import { ROUTES } from '../../constants/routes.enum';
 import { parseUrl } from '../../utils/routeUtils';
 import { OnBoardingAnalyticsEnum } from '../../pages/quick-start/consts';
 import { useSegment } from '../providers/SegmentProvider';
+import { useDigestDemoFlowContext } from './DigestDemoFlowProvider';
+
+const getOpacity = (id: string, hoveredHintId?: string, sequence?: { opacity: number }): number => {
+  if (hoveredHintId) {
+    return hoveredHintId === id ? 1 : 0.4;
+  }
+
+  return sequence?.opacity ?? 1;
+};
```

Comment:
this is how the opacity is calculated: 
- if we have set the `hoveredHintId` then the hovered item should have opacity 1 and others should have 0.4
- otherwise the animation sequence opacity should be applied